koptional-reactor-extensions/src/main/kotlin/com/gojuno/koptional/reactor/reactor.kt

```diff
@@ -0,0 +1,36 @@
+package com.gojuno.koptional.reactor
+
+import com.gojuno.koptional.None
+import com.gojuno.koptional.Optional
+import com.gojuno.koptional.Some
+import reactor.core.publisher.Flux
+
+private inline fun <reified R : Any> Flux<*>.ofType(): Flux<R> = ofType(R::class.java)
+
+/**
+ * Filters items emitted by a Flux by only emitting those that are `Some`.
+ * <p>
+ * <img width="640" height="310" src="https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/filter.png" alt="">
+ * <dl>
+ *  <dt><b>Scheduler:</b></dt>
+ *  <dd>{@code filter} does not operate by default on a particular {@link Scheduler}.</dd>
+ * </dl>
+ *
+ * @return a Flux that emits `Some.value` of those items emitted by the source Flux that are `Some`.
+ * @see <a href="http://reactivex.io/documentation/operators/filter.html">ReactiveX operators documentation: Filter</a>
```

Comment:
I guess we should use Reactor's pictures here https://raw.githubusercontent.com/reactor/reactor-core/v3.1.0.M2/src/docs/marble/filter.png